akka-stream/src/main/scala/akka/stream/stage/GraphStage.scala

```diff
@@ -1144,9 +1167,8 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
             case Initialized       ⇒ onAsyncInput(event, OptionVal.None)
           }
         }
-        case Completed ⇒ // do nothing here as stream is completed
       }
-      internalInvoke(event)
+      if (!stopped) internalInvoke(event)
```

Comment:
what happens if `stopped` is changed to `true` right after this check, i.e. when `internalInvoke` is called?

Reply:
`internalInvoke` is fire-and-forget regardless, so that's just a save-some-work-if-possible guard. 

The corresponding logic in `invokeWithPromise` is also a fail-a-bit-faster as it will check `stopped` after sending the invocation-with-promise as well.

Also, note that `stopped` isn't actually a state but the fact that the atomic ref `asyncCallbacksInProgress` is set to `null`.

Reply:
Is this in the hot path? Is it worth doing that then, since it adds atomic get to each invoke?

Reply:
I'll see if I can get rid of it without something breaking.